packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -416,15 +416,15 @@ export class EngineReportingAgent<TContext = any> {
       },
     ).catch((err: Error) => {
       throw new Error(
-        `Error sending report to Apollo Engine servers: ${err.message}`,
+        `Error sending report to Apollo Graph Manager servers: ${err.message}`,
```

Comment:
I think it's worth considering the rational which guides [the Node.js project itself](https://github.com/nodejs/node/blob/master/doc/guides/using-internal-errors.md#what-is-internalerrorsjs) here:

> Without a permanent identifier, userland code may need to inspect error messages to distinguish one error from another. An unfortunate result of that practice is that changes to error messages result in broken code in the ecosystem. For that reason, Node.js has considered error message changes to be breaking changes. 

I agree with this.  You can imagine someone aggregating errors or alerting on this and I believe us needing to change our test assertions within this PR is also a tell.

Bumping our major version isn't an option just yet, though Apollo Server 3.x is on the horizon and this package's deprecation is imminent.  Could we punt this?

Reply:
I can remove this, but I'm not positive if I agree with it being a breaking version change to Apollo Server. Some users may be parsing error messages directly for the part of our platform that communicates to BE services, but they might be doing the same with error messages that come from our back-end. Does that mean if we change the messages that accompany status codes, we should count that as a breaking change to Apollo Server as well? For instance, I've changed the messages that accompany `422` status codes several times to help users debug, but I don't think it should necessitate a breaking change.

Reply:
@zionts If codes (e.g. `422`) are involved, then it need not be a breaking change since it's possible for users to leverage those codes which are meant for machine consumption; this is the argument that the rational I linked to makes and the reason that Node.js is adding them as separate machine readable properties (i.e. `err.code`).

In the absence of codes, it is a potentially a breaking change as error messages are an external API surface area and they must rely on the value of `message` since we can't actually point users to a more stable approach.

As testing error conditions is usually wound into monitoring infrastructure of server consumers, changing this can — in a very tangible way — keep monitoring that those users have built from trigging if they've put a test or monitor in place that tests the part of this message that would be expected to remain the same.  e.g. Page admin if logs contain `/^Error sending reporting to Apollo Engine servers:/`.  

Of course, hopefully anyone who has put such a monitor/test in place would have likely identified the possible variability of it and left off the `${err.message}` part.  Also, using codes in our error messages will solve this, which we should absolutely do (I will make a note to make sure we do this).